### PR TITLE
Fix: scrub memory when unused

### PR DESCRIPTION
**Pull request by mchappell**
# Checklist

- [ ] JIRA - \<link>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**processReports**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6641/10340632670/index.html) for [d5c50ab0](https://github.com/input-output-hk/lace/pull/1360/commits/d5c50ab0db3295c163f4e6d14579a00966ffaf2b)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 26     | 6      | 0       | 0     | 32    | ❌     |

File: packages/core/src/ui/components/PasswordVerification/PasswordVerification.tsx [hunk 1]
```diff
@@ -16,12 +16,14 @@ export type PasswordVerificationProps = {
...
     isActive?: boolean;
   }>;
+  hasValue: boolean;
 } & PasswordProps;
 
 export const PasswordVerification = ({
   level,
   feedbacks,
   complexityBarList,
+  hasValue,
```
Comment (przemyslaw-wlodek) [inline packages/core/src/ui/components/PasswordVerification/PasswordVerification.tsx:26]: Should we name it better? 🤔 Proposals: `showComplexity`, `showFeedback`, `displayComplexity`, `complexityVisible` etc.

File: packages/core/src/ui/components/WalletSetup/WalletSetupConfirmationDialogProvider.tsx [hunk 2]
```diff
@@ -46,12 +48,13 @@ export const WalletSetupConfirmationDialogProvider = ({ children }: Props): Reac
           setIsDialogOpen(false);
           callback();
         };
+        clearSecrets();
```
Comment (przemyslaw-wlodek) [inline packages/core/src/ui/components/WalletSetup/WalletSetupConfirmationDialogProvider.tsx:51]: Let's discuss this one. I am not sure if I completely understand. This is called in the context of the `MultiWallet` component, only to confirm if the user wants to close the Modal. This has little to do with secrets per se but I presume this close button can be clicked at the moment when the user sees the Password input on the Dialog (onboarding flow).

File: packages/core/src/ui/components/WalletSetup/WalletSetupNamePasswordStep/WalletSetupNamePasswordStep.tsx [hunk 3]
```diff
@@ -115,17 +116,16 @@ export const WalletSetupNamePasswordStep = ({
...
         <PasswordVerification
           className={styles.input}
-          value={password}
           label={translations.passwordInputLabel}
           onChange={handlePasswordChange}
           level={score}
           feedbacks={passwordStrengthFeedbackMap[score] && [t(passwordStrengthFeedbackMap[score])]}
           complexityBarList={complexityBarList}
           data-testid="wallet-password-verification-input"
+          hasValue={!!password}
```
Comment (przemyslaw-wlodek) [inline packages/core/src/ui/components/WalletSetup/WalletSetupNamePasswordStep/WalletSetupNamePasswordStep.tsx:125]: Ok, I was confused at first :D These changes have no reflection in the app, it's a dead code. Nothing to do with it (we have a separate ticket), I'm leaving this comment for other reviewers.

File: packages/core/src/ui/components/WalletSetup/WalletSetupPasswordStep.tsx [hunk 4]
```diff
@@ -86,20 +87,19 @@ export const WalletSetupPasswordStep = ({
...
-          value={password}
           label={translations.password}
-          onChange={(e) => setPassword(e.target.value)}
+          onChange={(e) => setPassword(e.value)}
           level={score}
           feedbacks={feedbackTranslation}
           complexityBarList={complexityBarList}
           data-testid="wallet-setup-password-step-password"
           autoFocus
+          hasValue={!!password}
```
Comment (przemyslaw-wlodek) [inline packages/core/src/ui/components/WalletSetup/WalletSetupPasswordStep.tsx:97]: Dead code as well.

**Review by przemyslaw-wlodek — COMMENTED** [review]
Great work, thank you guys! 😄 No major comments, just some questions.

File: packages/core/src/ui/components/PasswordVerification/PasswordVerification.tsx (hunk 1 above)
Comment (mkazlauskas) [reply packages/core/src/ui/components/PasswordVerification/PasswordVerification.tsx]: Done in ee9d8bb46d3e360df51320b678a4b29f7132462c

File: packages/core/src/ui/components/WalletSetup/WalletSetupConfirmationDialogProvider.tsx (hunk 2 above)
Comment (mkazlauskas) [reply packages/core/src/ui/components/WalletSetup/WalletSetupConfirmationDialogProvider.tsx]: Done in 8cfb1bd9963ea00b10ca6418835d534958e1a978

**Review by wklos-iohk — APPROVED** [review]

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**